build.sbt

```diff
@@ -65,69 +65,89 @@ lazy val baseSettings = Seq(
   ),
   //dependencies
   resolvers ++= Resolvers.all,
-  libraryDependencies ++= Dependencies.common
+  libraryDependencies ++= Dependencies.common,
+  libraryDependencies ++= (CrossVersion.partialVersion(scalaVersion.value) match {
+    case Some((2, 12)) => Seq(Dependencies.scalaCollectionCompat)
+    case _             => Nil
+  })
 )
 
 lazy val compilePlugins = Seq(
-  addCompilerPlugin("org.typelevel" %% "kind-projector" % "0.10.3")
+  libraryDependencies ++= (CrossVersion.partialVersion(scalaVersion.value) match {
+    case Some((2, _)) => Seq(compilerPlugin(Dependencies.kindProjector))
```

Comment:
In `Dependencies` I'd like to create something like that:
```scala
object Dependencies{
 object Plugins {
   lazy val compilerPluginsFor2_13: Seq[ModuleID] = Seq(
     compilerPlugin("org.typelevel" %% "kind-projector" % "0.10.3" cross CrossVersion.binary)
   )

   lazy val compilerPluginsFor3: Seq[ModuleID] = Nil
 }
}
```

And use it in this way:
```scala
  libraryDependencies ++= ProjectDependencies.common ++ {
    CrossVersion.partialVersion(scalaVersion.value) match {
      case Some((2, 13)) => ProjectDependencies.Plugins.compilerPluginsFor2_13
      case Some((3, _)) => ProjectDependencies.Plugins.compilerPluginsFor3
      case _            => Nil
    }
  }
```